feat: Add review product screen ⭐

MarwaEbrahem opened the pull request.

MarwaEbrahem committed:
feat: add review products files

MarwaEbrahem committed:
Squashed commit of the following:

commit d4b43822855d488bfc4d8c2bac4dc0aa3e1545e3
Author: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>
Date:   Tue Aug 16 10:44:32 2022 +0200

    build: Add Cosmos View with a Bridge Stars View ⭐️ (#112)

    * add Cosmos package

    * add `StarsView`

MarwaEbrahem committed:
feat: Add review product UI and navigation bar

MarwaEbrahem committed:
Merge branch 'develop' into review-screen

MarwaEbrahem committed:
fix: fix ReviewProductNavigationBarBehavior file target

MarwaEbrahem committed:
fix: fix ReviewerTableViewCell file target

MarwaEbrahem committed:
fix: files target

MarwaEbrahem committed:
generate back image

MarwaEbrahem committed:
feat: Add mock product reviews files

MarwaEbrahem committed:
feat: apply clean architecture to get review data from network to app and update ui

MarwaEbrahem committed:
feat: update review cell with data

MarwaEbrahem committed:
feat: add skeleton view to review product screen

MarwaEbrahem committed:
fix: make height of table view cell dynamic to wrap content

MarwaEbrahem committed:
feat: Add error alert

MarwaEbrahem committed:
fix: edit naming convention

MarwaEbrahem committed:
fix: edit ui constrains

MarwaEbrahem committed:
fix: edit access level of UIAlertContoroller+Builder to public

MarwaEbrahem committed:
fix: Split review product view to subviews

MarwaEbrahem committed:
fix: enhance bind between reviewProduct view controller and view model

MarwaEbrahem committed:
fix: bind data between review product and rating details

ahmdmhasn commented on Dokan/Dokan/Scenes/ProductDetails/Views/ReviewerTableViewCell/ReviewerTableViewCell.swift:
```diff
@@ -5,34 +5,45 @@
+    let reviewerName: String
+    let reviewerComment: String
+    let rating: Double
 }
```
we've been using Nested ViewModel... Let's keep the convenience.

ahmdmhasn commented on Dokan/Dokan/Scenes/ProductDetails/Views/ReviewerTableViewCell/ReviewerTableViewCell.swift:
```diff
@@ -5,34 +5,45 @@
-    func configureCell(viewModel: ViewModel) {
-        #warning("Image will be configure after add kingfisher")
+    func configureCell(viewModel: ReviewerCellViewModel) {
+        reviewerImageView.setImage(with: viewModel.reviewerImageUrl, placeholderImage: UIImage(named: "star") ?? UIImage())
```
```suggestion
        reviewerImageView.setImage(with: viewModel.reviewerImageUrl, placeholderImage: .star)
```

ahmdmhasn commented on Dokan/Dokan/Scenes/ProductDetails/Views/ReviewerTableViewCell/ReviewerTableViewCell.swift:
```diff
@@ -5,34 +5,45 @@
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        contentView.frame = contentView.frame.inset(by: UIEdgeInsets(top: 0, left: 0, bottom: 20, right: 0))
```
Let's add this to `awakeFromNib` to make sure it performed only once.

ahmdmhasn commented on Core/Sources/Core/Models/DomainConvertible/Reviews.swift:
```diff
@@ -0,0 +1,26 @@
+        let reviewList = reviews.map { $0.toDomain() }
+
+        return Domain.Reviews(rating: rating, totalRatingNumber: totalRatingNumber, fiveStarRatingNumber: fiveStarRatingNumber, fourStarRatingNumber: fourStarRatingNumber, threeStarRatingNumber: threeStarRatingNumber, twoStarRatingNumber: twoStarRatingNumber, oneStarRatingNumber: oneStarRatingNumber, reviews: reviewList)
+    }
```
```suggestion
    func toDomain() -> Domain.Reviews {
        return Domain.Reviews(rating: rating, 
                                                 totalRatingNumber: totalRatingNumber, 
                                                 fiveStarRatingNumber: fiveStarRatingNumber, 
                                                 fourStarRatingNumber: fourStarRatingNumber, 
                                                 threeStarRatingNumber: threeStarRatingNumber, 
                                                 twoStarRatingNumber: twoStarRatingNumber, 
                                                 oneStarRatingNumber: oneStarRatingNumber, 
                                                 reviews: reviews.map { $0.toDomain() })
    }
```

ahmdmhasn commented on Domain/Sources/Domain/Repository/ReviewsRepository.swift:
```diff
@@ -0,0 +1,17 @@
+
+    /// Load Reviews
+    ///
+    func loadReviews(completion: @escaping (Domain.Reviews?, Error?) -> Void)
```
What if there are no reviews or errors? Let's use `Result<Domain.Reviews, Error>` instead

ahmdmhasn commented on Domain/Sources/Domain/Model/Reviews.swift:
```diff
@@ -0,0 +1,46 @@
+    public let twoStarRatingNumber, oneStarRatingNumber: Int
+    public let reviews: [Review]
+
+    public init(rating: Double, totalRatingNumber: Int, fiveStarRatingNumber: Int, fourStarRatingNumber: Int, threeStarRatingNumber: Int, twoStarRatingNumber: Int, oneStarRatingNumber: Int, reviews: [Review]) {
```
```suggestion
    public init(rating: Double, 
                      totalRatingNumber: Int, 
                      fiveStarRatingNumber: Int, 
                      fourStarRatingNumber: Int, 
                      threeStarRatingNumber: Int, 
                      twoStarRatingNumber: Int, 
                      oneStarRatingNumber: Int, 
                      reviews: [Review]) {
```

ahmdmhasn commented on Dokan/Dokan/Scenes/ReviewProduct/Views/RatingProgressDetails/RatingProgressDetails.swift:
```diff
@@ -0,0 +1,84 @@
+        super.init(coder: coder)
+        viewSetup()
+    }
+}
```
Read about it... Much easier and lesser code
```suggestion
class RatingProgressDetails: UIViewFromNib {

    // MARK: - Outlets

    @IBOutlet private(set) weak var contentView: UIView!
    @IBOutlet private(set) weak var starRating: StarsView!
    @IBOutlet private(set) weak var progressBarRating: UIProgressView!
    @IBOutlet private(set) weak var starsRatingLabel: UILabel!

    // MARK: - Properties

    var progressDetails: RatingProgressDetailsViewModel? {
        didSet {
            guard let progressDetails = progressDetails else {
                return
            }

            starRating.applyStyleToView(Double(progressDetails.starNumber))
            starsRatingLabel.text = String(progressDetails.rating)
            updateProgressBarRating(progressDetails: progressDetails)
        }
    }
}
```

ahmdmhasn commented on Dokan/Dokan/Scenes/ReviewProduct/Views/RatingProgressDetails/RatingProgressDetails.swift:
```diff
@@ -0,0 +1,84 @@
+    private enum Constants {
+        static let viewNibName = "RatingProgressDetails"
+    }
+}
```
Won't be needed. For other cases, you can use `"\(RatingProgressDetails.self)"` to make sure it changes when you rename the view.

ahmdmhasn commented on Dokan/Dokan/Scenes/ReviewProduct/Views/RatingProgressDetails/RatingProgressDetails.swift:
```diff
@@ -0,0 +1,84 @@
+    let rating: Int
+    let starNumber: Int
+    let totalRatingNumber: Int
+}
```
Let's use nested view model

ahmdmhasn commented on Dokan/Dokan/Scenes/ReviewProduct/Views/RatingDetails/RatingDetailsView.swift:
```diff
@@ -0,0 +1,92 @@
+import UIDokan
+import UIKit
+
+class RatingDetailsView: UIView {
```
```suggestion
class RatingDetailsView: UIViewFromNib {
```

ahmdmhasn commented on Dokan/Dokan/Scenes/ReviewProduct/ReviewProductViewModelType.swift:
```diff
@@ -0,0 +1,33 @@
+/// ReviewProduct ViewModel Output
+///
+protocol ReviewProductViewModelOutput {
+    func loadReviews()
```
Input

ahmdmhasn commented on Dokan/Dokan/Scenes/ReviewProduct/ReviewProductViewModelType.swift:
```diff
@@ -0,0 +1,33 @@
+    var showAlertClosure: () -> Void { get set }
+    var updateLoadingStatus: () -> Void { get set }
+    var showNavBarClosure: () -> Void { get set }
+    func getCellViewModel(at indexPath: IndexPath) -> ReviewerCellViewModel
```
output

ahmdmhasn commented on Dokan/Dokan/Scenes/ReviewProduct/ReviewProductViewModelType.swift:
```diff
@@ -0,0 +1,33 @@
+    var reloadTableViewClosure: () -> Void { get set }
+    var showAlertClosure: () -> Void { get set }
+    var updateLoadingStatus: () -> Void { get set }
+    var showNavBarClosure: () -> Void { get set }
```
```suggestion
    var onReloadData: () -> Void { get set } // We should not mention any UI related stuff since we can implement it using collection view or SwiftUI
    var onShowAlertClosure: (String) -> Void { get set } // We can remove the alert message.
    var onUpdateLoadingStatus: (Bool) -> Void { get set }
    var showNavBarClosure: () -> Void { get set } // Not descriptive
```

ahmdmhasn commented on Dokan/Dokan/Scenes/ReviewProduct/ReviewProductViewModel.swift:
```diff
@@ -0,0 +1,144 @@
+
+// MARK: - states
+
+enum State {
```
This is a specific view state, let's add it to `ReviewProductViewModelType.swift` and rename it. 
```suggestion
enum ReviewProductState {
```

ahmdmhasn commented on Dokan/Dokan/Scenes/ReviewProduct/ReviewProductViewModel.swift:
```diff
@@ -0,0 +1,144 @@
+        for review in reviewsList {
+            cellVMs.append(createCellViewModel(reviews: review))
+        }
+        cellViewModels = cellVMs
```
```suggestion
        cellViewModels = reviewsList.map { createCellViewModel(reviews: $0) }
```

ahmdmhasn commented on Dokan/Dokan/Scenes/ReviewProduct/ReviewProductViewModel.swift:
```diff
@@ -0,0 +1,144 @@
+    // MARK: - Properties
+
+    private let repository: ReviewsRepository
+    private var reviewsList: [Domain.Review] = .init()
```
```suggestion
    private var reviewsList: [Domain.Review] = []
```

ahmdmhasn commented on Dokan/Dokan/Scenes/ReviewProduct/ReviewProductViewModel.swift:
```diff
@@ -0,0 +1,144 @@
+    private let repository: ReviewsRepository
+    private var reviewsList: [Domain.Review] = .init()
+
+    private var cellViewModels: [ReviewerCellViewModel] = .init() {
```
```suggestion
    private var cellViewModels: [ReviewerCellViewModel] = [] {
```

ahmdmhasn commented on Dokan/Dokan/Scenes/ReviewProduct/ReviewProductViewModel.swift:
```diff
@@ -0,0 +1,144 @@
+        }
+    }
+
+    var alertMessage: String = .init() {
```
```suggestion
    var alertMessage: String = "" {
```

ahmdmhasn commented on Dokan/Dokan/Scenes/ReviewProduct/ReviewProductViewModel.swift:
```diff
@@ -0,0 +1,144 @@
+        }
+    }
+
+    var navBarRating: Double = .init() {
```
```suggestion
    var navBarRating: Double = .zero {
```

ahmdmhasn commented on Dokan/Dokan/Scenes/ReviewProduct/ReviewProductNavigationBarBehavior.swift:
```diff
@@ -0,0 +1,64 @@
+        // MARK: Add back button to navigation bar
+
+        self.onBack = onBack
+        let backButton = UIBarButtonItem(image: UIImage(named: "left")?.withRenderingMode(.alwaysOriginal),
```
```suggestion
        let backButton = UIBarButtonItem(image: UIImage.left.withRenderingMode(.alwaysOriginal),
```

ahmdmhasn commented on Dokan/Dokan/Scenes/ReviewProduct/ReviewProductNavigationBarBehavior.swift:
```diff
@@ -0,0 +1,64 @@
+
+        // MARK: Add star image and rating label
+
+        lazy var ratingStackView: UIStackView = {
```
No need for lazy since it should be used in the next line. 
```suggestion
        let ratingStackView: UIStackView = {
```

ahmdmhasn commented on Dokan/Dokan/Scenes/ReviewProduct/ReviewProductNavigationBarBehavior.swift:
```diff
@@ -0,0 +1,64 @@
+            ratingLabel.text = String(rating)
+            ratingLabel.textColor = .black
+            let starImage = UIImageView()
+            starImage.image = UIImage(named: "star")
```
```suggestion
            starImage.image = .star
```

ahmdmhasn commented on Networking/Sources/Networking/Remote/MockNetwork.swift:
```diff
@@ -0,0 +1,130 @@
+
+    private enum Constants {
+
+        static let mockProductReviewsJsonResponse: [String: Any] = [
```
Let's move this json to a file a try to decode it instead of making it hard coded here. You can read about load data from file. It needs only one line of code.

ahmdmhasn commented on Networking/Sources/Networking/Models/Reviews.swift:
```diff
@@ -0,0 +1,27 @@
+
+// MARK: - Reviews
+
+public struct Reviews: Codable {
+    public let rating: Double
+    public let totalRatingNumber, fiveStarRatingNumber, fourStarRatingNumber, threeStarRatingNumber: Int
+    public let twoStarRatingNumber, oneStarRatingNumber: Int
+    public let reviews: [Review]
+}
+
+// MARK: - Review
+
+public struct Review: Codable {
```
```suggestion
public struct Reviews: Decodable {
    public let rating: Double
    public let totalRatingNumber, fiveStarRatingNumber, fourStarRatingNumber, threeStarRatingNumber: Int
    public let twoStarRatingNumber, oneStarRatingNumber: Int
    public let reviews: [Review]
}

// MARK: - Review

public struct Review: Decodable {
```

ahmdmhasn commented on UIDokan/Sources/UIDokan/ReusableViews/StarsView.swift:
```diff
@@ -7,7 +7,19 @@
-public class StarsView: CosmosView {}
+public class StarsView: CosmosView {
+
+    public func applyStyleToView(_ rate: Double) {
```
Violates single responsibility. Let's create one for updating the rate and another for apply style.

ahmdmhasn commented on Dokan/Dokan/Scenes/ReviewProduct/ReviewProductViewController.swift:
```diff
@@ -0,0 +1,196 @@
+extension ReviewProductViewController {
+
+    private enum Constants {
+        static let tableViewCellName = "ReviewerTableViewCell"
+        static let cellReuseIdentifier = "ReviewerTableViewCell"
```
We've helpers for register and dequeue so there's no reason for hardcoded values.

ahmdmhasn commented on Dokan/Dokan/Scenes/ReviewProduct/ReviewProductViewController.swift:
```diff
@@ -0,0 +1,196 @@
+    func bindViewModel() {
+
+        viewModel.showAlertClosure = { () in
+            DispatchQueue.main.async { [weak self] in
+                guard let self = self else { return }
+
+                let message = self.viewModel.alertMessage
+                self.showErrorAlert(error: message)
+            }
+        }
+
+        viewModel.showNavBarClosure = { () in
+            DispatchQueue.main.async { [weak self] in
+                guard let self = self else { return }
+
+                let rating = self.viewModel.navBarRating
+                self.configureNavBar(rating: rating)
+            }
```
Let's not used `DispatchQueue.main.async` unless it's truly needed. This will affect unit testing.

ahmdmhasn commented on Dokan/Dokan/Scenes/ReviewProduct/ReviewProductViewController.swift:
```diff
@@ -0,0 +1,196 @@
+
+    private var viewModel: ReviewProductViewModelType
+    private var navigationBarBehavior: ReviewProductNavigationBarBehavior?
+    private var reviewCount: Int = 10
+    private var loadingCell: Bool = true
```
Any logic should be added to the view model.

ahmdmhasn reviewed (commented):
Greeeat effort 🎉🔥

MarwaEbrahem committed:
fix: edit toDomain function

Co-authored-by: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>

MarwaEbrahem committed:
fix: change codable to decodable in model

Co-authored-by: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>

MarwaEbrahem committed:
Update Reviews init function style

Co-authored-by: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>

MarwaEbrahem committed:
fix: edit processFetchedReviews function

Co-authored-by: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>

MarwaEbrahem committed:
fix: Update initialize of reviewList variable

Co-authored-by: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>

MarwaEbrahem committed:
fix: Update initialize of cellViewModels variable

Co-authored-by: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>

MarwaEbrahem committed:
fix: Update initialize of alertMessage variable

Co-authored-by: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>

MarwaEbrahem committed:
fix: Update initialize of navBarRating variable

Co-authored-by: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>

MarwaEbrahem committed:
Update Dokan/Dokan/Scenes/ReviewProduct/ReviewProductNavigationBarBehavior.swift

Co-authored-by: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>

MarwaEbrahem committed:
fix: update state enum name

Co-authored-by: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>

MarwaEbrahem committed:
fix: update code

MarwaEbrahem committed:
fix: Update ReviewerTableViewCell

MarwaEbrahem committed:
fix: refactor RatingDetailsViewModel

MarwaEbrahem committed:
fix: refactor ReviewProductNavigationBarBehavior

MarwaEbrahem committed:
fix: update register ReviewerTableViewCell

MarwaEbrahem committed:
fix: refactor MockNetwork

MarwaEbrahem committed:
Squashed commit of the following:

commit 464666383afe8f3f70d17d4e1e9030ac5989482a
Author: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>
Date:   Mon Sep 5 05:46:54 2022 +0200

    fix: UIViewFromNib Crash When Used With any Custom View 👻 (#123)

    * fix typo in `bundle`

    * Update access modifiers

    * fix issue when using `UIViewFromNib`

commit f9644ceb638a5b724412bb533d588ec454350c5a
Author: RaniaZeid <67391043+RaniaZeid@users.noreply.github.com>
Date:   Thu Sep 1 13:31:29 2022 -0700

    feat: Get All Categories List to Category Remote  (#114)

    * create all category remote file

    * rename all category protocol and class file

commit 840d226f9d574a6f950f6ef745528988a0ce56d8
Author: Ahmed Nagy <ahmednaagy97@gmail.com>
Date:   Sun Aug 21 23:23:23 2022 +0200

    fix: buttonsView compatibility with iOS 13 and above 📈 (#106)

    * add buttons view

    * fix ButtonsView not showing and change the favorite button color

    * delete the test code in SceneDelegate

    * change favorite button color

    * make product details screen the root viewController

    code added to SceneDelegate to launch the Product Details screen on launch to test out current tasks

    * fix: make buttonsView compatible starting from iOS 13 and above

    * adding access control to ButtonsView

    * resolving minor issues regarding last PR comments

    * fixing naming issues

    * fix: sellerImageProfile outlets connections

    * fix InfoSellerView outlets connections

commit e9d38ab9abb078b4bda2eef38aea9f8036524862
Merge: ed39eef3 21d6b271
Author: claramounir <107275541+claramounir@users.noreply.github.com>
Date:   Sat Aug 20 13:11:34 2022 +0200

    Merge pull request #65 from motoon-eg/feat/product-details-reviews-view

    feat: Add reviews section to product details page 🌠

commit 21d6b2711ba45a4805cc9a4d04cf6056a626a592
Merge: f6ff7837 ed39eef3
Author: claramounir <claramounir555@gmail.com>
Date:   Sat Aug 20 12:53:53 2022 +0200

    Merge branch 'develop' into feat/product-details-reviews-view

    # Conflicts:
    #	Dokan/Dokan.xcodeproj/project.pbxproj
    #	Dokan/Dokan/Scenes/ProductDetails/Views/ReviewerTableViewCell/ReviewerTableViewCell.swift

commit f6ff783792814f51189c47d9ab71eca89541e70e
Author: claramounir <claramounir555@gmail.com>
Date:   Fri Aug 19 18:11:06 2022 +0200

    add reviews view

commit 3783ebd324935df55a5a9d05aa1f9fed2d38b1cb
Author: claramounir <claramounir555@gmail.com>
Date:   Fri Aug 19 18:07:21 2022 +0200

    add UIView+Extension

commit 8b9f9218e68f30efc5d945a49d12a570d3f3551b
Author: claramounir <claramounir555@gmail.com>
Date:   Sun Aug 7 01:30:24 2022 +0200

    delete duplicate files

commit daeb336a950df7c9ca81c1108b7522a5902a13ad
Merge: c8dbddcf 6a5cbfbc
Author: claramounir <claramounir555@gmail.com>
Date:   Sun Aug 7 00:39:31 2022 +0200

    Merge branch 'feat/product-details-reviews-view' of https://github.com/motoon-eg/dokan-store into feat/product-details-reviews-view

commit c8dbddcf4008cc2696b4b784944a87db826a4ae1
Author: claramounir <claramounir555@gmail.com>
Date:   Sun Aug 7 00:35:53 2022 +0200

    reslove duplicate build error

commit 6a5cbfbc707cee8e3c6bac92dd26bd844aca4677
Merge: d1817094 288c6c58
Author: claramounir <107275541+claramounir@users.noreply.github.com>
Date:   Sun Aug 7 00:16:08 2022 +0200

    Merge branch 'develop' into feat/product-details-reviews-view

commit d18170949d4ef49d0167f851c2586b39341c23ed
Author: claramounir <claramounir555@gmail.com>
Date:   Sun Aug 7 00:04:48 2022 +0200

    solve conflict

commit fd77eb5a9566b220d7296deb1afababbdea27535
Author: claramounir <107275541+claramounir@users.noreply.github.com>
Date:   Sat Aug 6 07:58:23 2022 +0200

    Update ReviewsView.swift

commit 5e6d6554a6e05d3771a8907681473c136e349f7e
Author: claramounir <107275541+claramounir@users.noreply.github.com>
Date:   Sat Aug 6 07:54:11 2022 +0200

    Update Dokan/Dokan/Scenes/ProductDetails/ReviewsView.swift

    Co-authored-by: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>

commit ab944f9313c4c366ea6696039c128151e68f55a6
Author: claramounir <107275541+claramounir@users.noreply.github.com>
Date:   Sat Aug 6 07:53:50 2022 +0200

    Update Dokan/Dokan/Scenes/ProductDetails/Cell/ReviewerTableViewCell.swift

    Co-authored-by: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>

commit 8a6581d8b2bdbc4759183203ff657d5a4ccb71bb
Author: claramounir <107275541+claramounir@users.noreply.github.com>
Date:   Sat Aug 6 07:52:07 2022 +0200

    Update ReviewerTableViewCell.swift

commit e5043993868446deb50cffe483096ed0b8d8063d
Author: claramounir <107275541+claramounir@users.noreply.github.com>
Date:   Sat Aug 6 07:45:26 2022 +0200

    Update ReviewerTableViewCell.swift

commit 068aa17882e14ccf5e9f53c917e8d25cd3918c92
Author: claramounir <107275541+claramounir@users.noreply.github.com>
Date:   Sat Aug 6 07:38:54 2022 +0200

    Update Dokan/Dokan/Scenes/ProductDetails/Cell/ReviewerTableViewCell.swift

    Co-authored-by: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>

commit de18a8ffb105e852a4b309ff3ed650e30d752b8f
Author: claramounir <107275541+claramounir@users.noreply.github.com>
Date:   Sat Aug 6 07:38:34 2022 +0200

    Update ReviewerTableViewCell.swift

commit e7266756d96082c2d741b4aca2bbc16666914f0c
Author: claramounir <107275541+claramounir@users.noreply.github.com>
Date:   Sat Aug 6 07:37:30 2022 +0200

    Update ReviewerTableViewCell.swift

commit f50ab4dc7ea6643c9b23571e36df582751fc592e
Merge: 73b6bcf3 6ba5dca3
Author: claramounir <107275541+claramounir@users.noreply.github.com>
Date:   Sat Aug 6 07:35:20 2022 +0200

    Merge branch 'develop' into feat/product-details-reviews-view

commit 73b6bcf3f672e177bd9725107fef4bcb009e8170
Author: Ahmed M. Hassan <ahmdmhasn@gmail.com>
Date:   Sat Aug 6 01:35:59 2022 +0200

    Fix indentation

commit 3a4a656389d37b36f699a62e70cadafeec9ebf16
Author: Ahmed M. Hassan <ahmdmhasn@gmail.com>
Date:   Sat Aug 6 01:31:16 2022 +0200

    Fix conflicts

commit e839b0a4f635a2a85c948fd405a9953a65075ca9
Author: claramounir <claramounir555@gmail.com>
Date:   Sun Jul 31 22:41:26 2022 +0200

    ui changes

commit 786850b3b40ddba0fa46ac509e2aae3ef9dd2a1b
Author: RaniaZeid <67391043+RaniaZeid@users.noreply.github.com>
Date:   Thu Jul 28 14:25:28 2022 -0700

    create reviwer Table View Nib file

commit 437cbea8dc6d42989210af2272348f25bc795adc
Author: claramounir <claramounir555@gmail.com>
Date:   Thu Jul 28 19:47:24 2022 +0200

    restore reviews view

commit 73289b504bacaffa01afaf4838a41e71794b17f8
Author: claramounir <claramounir555@gmail.com>
Date:   Thu Jul 28 19:34:11 2022 +0200

    solve review notes

commit 514382c2d4bc4522dc59828d386b8b3d896f73ad
Author: claramounir <claramounir555@gmail.com>
Date:   Wed Jul 20 21:45:40 2022 +0200

    add review view design  :v:

commit ab0ee749f1ff5ecc74e8a834c4a94cc78a2d19f5
Merge: 34ddf7e7 c08c5b29
Author: Ahmed M. Hassan <ahmdmhasn@gmail.com>
Date:   Sat Aug 6 01:11:29 2022 +0200

    Merge remote-tracking branch 'origin/develop' into feat/product-details-reviews-view

    # Conflicts:
    #	Dokan/Dokan/Assets.xcassets/star.imageset/star.png

commit 34ddf7e74a4b4ec4e449e6a56be0eaadcfd470b7
Merge: cfe40eb5 a12ae42a
Author: claramounir <107275541+claramounir@users.noreply.github.com>
Date:   Thu Aug 4 12:42:29 2022 +0200

    Merge branch 'develop' into feat/product-details-reviews-view

commit cfe40eb5961904b28f74893738703cc2325a9b2e
Merge: e0847892 d258362d
Author: claramounir <107275541+claramounir@users.noreply.github.com>
Date:   Wed Aug 3 22:07:53 2022 +0200

    Merge branch 'develop' into feat/product-details-reviews-view

commit e0847892463df34b5ca886cf9ca15a2fc2b73976
Author: claramounir <claramounir555@gmail.com>
Date:   Tue Aug 2 16:12:33 2022 +0200

    configure review cell

commit 3023d59314659dbd92b6e1e5f1eb73acad0f0123
Author: claramounir <claramounir555@gmail.com>
Date:   Sun Jul 31 22:41:26 2022 +0200

    ui changes

commit 9d1556dfe6b7b53213bbf21fa59004e8e92e2b4b
Author: claramounir <claramounir555@gmail.com>
Date:   Fri Jul 29 15:53:59 2022 +0200

    solve conflict

commit c2f35cf71523af33cf1c0226e5505b750b51fdb8
Merge: 3845ac22 f661127e
Author: claramounir <claramounir555@gmail.com>
Date:   Fri Jul 29 01:09:18 2022 +0200

    no message

commit 3845ac2233487d5387359320ce4f461ee28d0d16
Author: claramounir <claramounir555@gmail.com>
Date:   Fri Jul 29 01:03:28 2022 +0200

    reviews view tests

commit f661127e4309b5673bd3f69d28dce1a6a9b6c399
Author: RaniaZeid <67391043+RaniaZeid@users.noreply.github.com>
Date:   Thu Jul 28 14:50:21 2022 -0700

    add outlets and cell model

commit 307a54e280117966641dd2f309fc3ecbe8bbcaf9
Author: RaniaZeid <67391043+RaniaZeid@users.noreply.github.com>
Date:   Thu Jul 28 14:44:36 2022 -0700

    create reviewer cell design

commit 47e57bbe87726e2756a7e75d0c917b8981d4320a
Author: RaniaZeid <67391043+RaniaZeid@users.noreply.github.com>
Date:   Thu Jul 28 14:25:28 2022 -0700

    create reviwer Table View Nib file

commit e799b756eff230a3cb8cb9fc96b3be26796ccfbc
Author: claramounir <claramounir555@gmail.com>
Date:   Thu Jul 28 19:47:24 2022 +0200

    restore reviews view

commit 663a7dee17fbb3b217c2532ad4b035a68d49afa8
Author: claramounir <claramounir555@gmail.com>
Date:   Thu Jul 28 19:34:11 2022 +0200

    solve review notes

commit 68c7eb8358218c23593362abe14d9f4677d85a84
Merge: 597352ee 575cf24d
Author: Ahmed M. Hassan <ahmdmhasn@gmail.com>
Date:   Mon Jul 25 02:20:16 2022 +0200

    Merge remote-tracking branch 'origin/develop' into feat/product-details-reviews-view

    # Conflicts:
    #	Dokan/Dokan.xcodeproj/project.pbxproj

commit 597352ee002ef1e2f12a445df801d6994bbfcc8b
Author: claramounir <107275541+claramounir@users.noreply.github.com>
Date:   Thu Jul 21 18:01:49 2022 +0200

    Update ReviewsView.swift

commit b63ce257da8ce17c01d355399177ce97321e7704
Author: claramounir <claramounir555@gmail.com>
Date:   Wed Jul 20 21:45:40 2022 +0200

    add review view design  :v:

commit 8312a23a121efe491e164a7abf550eef85747787
Merge: ea116fc5 de202755
Author: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>
Date:   Sun Jul 17 11:07:55 2022 +0200

    Merge branch 'develop' into feat/updateswiftversionto5.5

commit ea116fc52ff9e142673c1da8d751cc4cbc061488
Merge: 6f035770 a581bb94
Author: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>
Date:   Sun Jul 17 11:02:47 2022 +0200

    Merge branch 'develop' into feat/updateswiftversionto5.5

commit 6f035770ba83a29d83ec2b3297677f1a800e6bda
Merge: d751c91e 5cfba698
Author: RaniaZeid <67391043+RaniaZeid@users.noreply.github.com>
Date:   Sat Jul 16 20:33:47 2022 +0200

    Merge pull request #60 from motoon-eg/feat/updateverificationscene

    add action to resend and mainbutton

commit 5cfba698e79661ef02762f8829eae1ff1d3a585c
Author: RaniaZeid <67391043+RaniaZeid@users.noreply.github.com>
Date:   Sat Jul 16 11:31:19 2022 -0700

    add action to resend and mainbutton

commit d751c91eaaca210a61d78a43483e51683f5809fb
Author: Ahmad Yasser <42544598+ahmdyasser@users.noreply.github.com>
Date:   Sat Jul 16 19:41:00 2022 +0200

    make OTCTextfield class open (#59)

commit b3339a93105e9e98fae67c96f2486bbb4d7e783f
Author: RaniaZeid <67391043+RaniaZeid@users.noreply.github.com>
Date:   Sat Jul 16 06:40:49 2022 -0700

    update swift version at domain and core

MarwaEbrahem committed:
feat: solve merge conflict

MarwaEbrahem committed:
fix: setup view by using uiviewfromnib file

MarwaEbrahem committed:
fix: fix import module in test files

MarwaEbrahem committed:
Merge branch 'develop' into review-screen

MarwaEbrahem committed:
feat: Add unit test to review product files (#122)

* feat: Add unit test to review product files

* fix: refactor test code

* fix: add mock view model to project

* Squashed commit of the following:

commit fb44a8bfcb0416047a2c5a04bbf7128bbf1d8674
Merge: 48904c65 46466638
Author: MarwaEbrahem <31275340+MarwaEbrahem@users.noreply.github.com>
Date:   Mon Sep 5 18:01:28 2022 +0200

    Merge branch 'develop' into review-screen

commit 464666383afe8f3f70d17d4e1e9030ac5989482a
Author: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>
Date:   Mon Sep 5 05:46:54 2022 +0200

    fix: UIViewFromNib Crash When Used With any Custom View 👻 (#123)

    * fix typo in `bundle`

    * Update access modifiers

    * fix issue when using `UIViewFromNib`

commit 48904c6516e48656cdfcdf2d5e24e462afa08c98
Author: MarwaEbrahem <marwa15shanab@gmail.com>
Date:   Sun Sep 4 16:52:34 2022 -0700

    fix: setup view by using uiviewfromnib file

commit a16934eb13ef8421a2c5ed92aed428a1f26c75da
Author: MarwaEbrahem <marwa15shanab@gmail.com>
Date:   Sun Sep 4 16:37:51 2022 -0700

    feat: solve merge conflict

commit 03a6d5f870618b7945dd1a63b48c9fbfd86d4f05
Author: MarwaEbrahem <marwa15shanab@gmail.com>
Date:   Sun Sep 4 16:16:31 2022 -0700

    Squashed commit of the following:

    commit 464666383afe8f3f70d17d4e1e9030ac5989482a
    Author: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>
    Date:   Mon Sep 5 05:46:54 2022 +0200

        fix: UIViewFromNib Crash When Used With any Custom View 👻 (#123)

        * fix typo in `bundle`

        * Update access modifiers

        * fix issue when using `UIViewFromNib`

    commit f9644ceb638a5b724412bb533d588ec454350c5a
    Author: RaniaZeid <67391043+RaniaZeid@users.noreply.github.com>
    Date:   Thu Sep 1 13:31:29 2022 -0700

        feat: Get All Categories List to Category Remote  (#114)

        * create all category remote file

        * rename all category protocol and class file

    commit 840d226f9d574a6f950f6ef745528988a0ce56d8
    Author: Ahmed Nagy <ahmednaagy97@gmail.com>
    Date:   Sun Aug 21 23:23:23 2022 +0200

        fix: buttonsView compatibility with iOS 13 and above 📈 (#106)

        * add buttons view

        * fix ButtonsView not showing and change the favorite button color

        * delete the test code in SceneDelegate

        * change favorite button color

        * make product details screen the root viewController

        code added to SceneDelegate to launch the Product Details screen on launch to test out current tasks

        * fix: make buttonsView compatible starting from iOS 13 and above

        * adding access control to ButtonsView

        * resolving minor issues regarding last PR comments

        * fixing naming issues

        * fix: sellerImageProfile outlets connections

        * fix InfoSellerView outlets connections

    commit e9d38ab9abb078b4bda2eef38aea9f8036524862
    Merge: ed39eef3 21d6b271
    Author: claramounir <107275541+claramounir@users.noreply.github.com>
    Date:   Sat Aug 20 13:11:34 2022 +0200

        Merge pull request #65 from motoon-eg/feat/product-details-reviews-view

        feat: Add reviews section to product details page 🌠

    commit 21d6b2711ba45a4805cc9a4d04cf6056a626a592
    Merge: f6ff7837 ed39eef3
    Author: claramounir <claramounir555@gmail.com>
    Date:   Sat Aug 20 12:53:53 2022 +0200

        Merge branch 'develop' into feat/product-details-reviews-view

        # Conflicts:
        #	Dokan/Dokan.xcodeproj/project.pbxproj
        #	Dokan/Dokan/Scenes/ProductDetails/Views/ReviewerTableViewCell/ReviewerTableViewCell.swift

    commit f6ff783792814f51189c47d9ab71eca89541e70e
    Author: claramounir <claramounir555@gmail.com>
    Date:   Fri Aug 19 18:11:06 2022 +0200

        add reviews view

    commit 3783ebd324935df55a5a9d05aa1f9fed2d38b1cb
    Author: claramounir <claramounir555@gmail.com>
    Date:   Fri Aug 19 18:07:21 2022 +0200

        add UIView+Extension

    commit 8b9f9218e68f30efc5d945a49d12a570d3f3551b
    Author: claramounir <claramounir555@gmail.com>
    Date:   Sun Aug 7 01:30:24 2022 +0200

        delete duplicate files

    commit daeb336a950df7c9ca81c1108b7522a5902a13ad
    Merge: c8dbddcf 6a5cbfbc
    Author: claramounir <claramounir555@gmail.com>
    Date:   Sun Aug 7 00:39:31 2022 +0200

        Merge branch 'feat/product-details-reviews-view' of https://github.com/motoon-eg/dokan-store into feat/product-details-reviews-view

    commit c8dbddcf4008cc2696b4b784944a87db826a4ae1
    Author: claramounir <claramounir555@gmail.com>
    Date:   Sun Aug 7 00:35:53 2022 +0200

        reslove duplicate build error

    commit 6a5cbfbc707cee8e3c6bac92dd26bd844aca4677
    Merge: d1817094 288c6c58
    Author: claramounir <107275541+claramounir@users.noreply.github.com>
    Date:   Sun Aug 7 00:16:08 2022 +0200

        Merge branch 'develop' into feat/product-details-reviews-view

    commit d18170949d4ef49d0167f851c2586b39341c23ed
    Author: claramounir <claramounir555@gmail.com>
    Date:   Sun Aug 7 00:04:48 2022 +0200

        solve conflict

    commit fd77eb5a9566b220d7296deb1afababbdea27535
    Author: claramounir <107275541+claramounir@users.noreply.github.com>
    Date:   Sat Aug 6 07:58:23 2022 +0200

        Update ReviewsView.swift

    commit 5e6d6554a6e05d3771a8907681473c136e349f7e
    Author: claramounir <107275541+claramounir@users.noreply.github.com>
    Date:   Sat Aug 6 07:54:11 2022 +0200

        Update Dokan/Dokan/Scenes/ProductDetails/ReviewsView.swift

        Co-authored-by: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>

    commit ab944f9313c4c366ea6696039c128151e68f55a6
    Author: claramounir <107275541+claramounir@users.noreply.github.com>
    Date:   Sat Aug 6 07:53:50 2022 +0200

        Update Dokan/Dokan/Scenes/ProductDetails/Cell/ReviewerTableViewCell.swift

        Co-authored-by: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>

    commit 8a6581d8b2bdbc4759183203ff657d5a4ccb71bb
    Author: claramounir <107275541+claramounir@users.noreply.github.com>
    Date:   Sat Aug 6 07:52:07 2022 +0200

        Update ReviewerTableViewCell.swift

    commit e5043993868446deb50cffe483096ed0b8d8063d
    Author: claramounir <107275541+claramounir@users.noreply.github.com>
    Date:   Sat Aug 6 07:45:26 2022 +0200

        Update ReviewerTableViewCell.swift

    commit 068aa17882e14ccf5e9f53c917e8d25cd3918c92
    Author: claramounir <107275541+claramounir@users.noreply.github.com>
    Date:   Sat Aug 6 07:38:54 2022 +0200

        Update Dokan/Dokan/Scenes/ProductDetails/Cell/ReviewerTableViewCell.swift

        Co-authored-by: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>

    commit de18a8ffb105e852a4b309ff3ed650e30d752b8f
    Author: claramounir <107275541+claramounir@users.noreply.github.com>
    Date:   Sat Aug 6 07:38:34 2022 +0200

        Update ReviewerTableViewCell.swift

    commit e7266756d96082c2d741b4aca2bbc16666914f0c
    Author: claramounir <107275541+claramounir@users.noreply.github.com>
    Date:   Sat Aug 6 07:37:30 2022 +0200

        Update ReviewerTableViewCell.swift

    commit f50ab4dc7ea6643c9b23571e36df582751fc592e
    Merge: 73b6bcf3 6ba5dca3
    Author: claramounir <107275541+claramounir@users.noreply.github.com>
    Date:   Sat Aug 6 07:35:20 2022 +0200

        Merge branch 'develop' into feat/product-details-reviews-view

    commit 73b6bcf3f672e177bd9725107fef4bcb009e8170
    Author: Ahmed M. Hassan <ahmdmhasn@gmail.com>
    Date:   Sat Aug 6 01:35:59 2022 +0200

        Fix indentation

    commit 3a4a656389d37b36f699a62e70cadafeec9ebf16
    Author: Ahmed M. Hassan <ahmdmhasn@gmail.com>
    Date:   Sat Aug 6 01:31:16 2022 +0200

        Fix conflicts

    commit e839b0a4f635a2a85c948fd405a9953a65075ca9
    Author: claramounir <claramounir555@gmail.com>
    Date:   Sun Jul 31 22:41:26 2022 +0200

        ui changes

    commit 786850b3b40ddba0fa46ac509e2aae3ef9dd2a1b
    Author: RaniaZeid <67391043+RaniaZeid@users.noreply.github.com>
    Date:   Thu Jul 28 14:25:28 2022 -0700

        create reviwer Table View Nib file

    commit 437cbea8dc6d42989210af2272348f25bc795adc
    Author: claramounir <claramounir555@gmail.com>
    Date:   Thu Jul 28 19:47:24 2022 +0200

        restore reviews view

    commit 73289b504bacaffa01afaf4838a41e71794b17f8
    Author: claramounir <claramounir555@gmail.com>
    Date:   Thu Jul 28 19:34:11 2022 +0200

        solve review notes

    commit 514382c2d4bc4522dc59828d386b8b3d896f73ad
    Author: claramounir <claramounir555@gmail.com>
    Date:   Wed Jul 20 21:45:40 2022 +0200

        add review view design  :v:

    commit ab0ee749f1ff5ecc74e8a834c4a94cc78a2d19f5
    Merge: 34ddf7e7 c08c5b29
    Author: Ahmed M. Hassan <ahmdmhasn@gmail.com>
    Date:   Sat Aug 6 01:11:29 2022 +0200

        Merge remote-tracking branch 'origin/develop' into feat/product-details-reviews-view

        # Conflicts:
        #	Dokan/Dokan/Assets.xcassets/star.imageset/star.png

    commit 34ddf7e74a4b4ec4e449e6a56be0eaadcfd470b7
    Merge: cfe40eb5 a12ae42a
    Author: claramounir <107275541+claramounir@users.noreply.github.com>
    Date:   Thu Aug 4 12:42:29 2022 +0200

        Merge branch 'develop' into feat/product-details-reviews-view

    commit cfe40eb5961904b28f74893738703cc2325a9b2e
    Merge: e0847892 d258362d
    Author: claramounir <107275541+claramounir@users.noreply.github.com>
    Date:   Wed Aug 3 22:07:53 2022 +0200

        Merge branch 'develop' into feat/product-details-reviews-view

    commit e0847892463df34b5ca886cf9ca15a2fc2b73976
    Author: claramounir <claramounir555@gmail.com>
    Date:   Tue Aug 2 16:12:33 2022 +0200

        configure review cell

    commit 3023d59314659dbd92b6e1e5f1eb73acad0f0123
    Author: claramounir <claramounir555@gmail.com>
    Date:   Sun Jul 31 22:41:26 2022 +0200

        ui changes

    commit 9d1556dfe6b7b53213bbf21fa59004e8e92e2b4b
    Author: claramounir <claramounir555@gmail.com>
    Date:   Fri Jul 29 15:53:59 2022 +0200

        solve conflict

    commit c2f35cf71523af33cf1c0226e5505b750b51fdb8
    Merge: 3845ac22 f661127e
    Author: claramounir <claramounir555@gmail.com>
    Date:   Fri Jul 29 01:09:18 2022 +0200

        no message

    commit 3845ac2233487d5387359320ce4f461ee28d0d16
    Author: claramounir <claramounir555@gmail.com>
    Date:   Fri Jul 29 01:03:28 2022 +0200

        reviews view tests

    commit f661127e4309b5673bd3f69d28dce1a6a9b6c399
    Author: RaniaZeid <67391043+RaniaZeid@users.noreply.github.com>
    Date:   Thu Jul 28 14:50:21 2022 -0700

        add outlets and cell model

    commit 307a54e280117966641dd2f309fc3ecbe8bbcaf9
    Author: RaniaZeid <67391043+RaniaZeid@users.noreply.github.com>
    Date:   Thu Jul 28 14:44:36 2022 -0700

        create reviewer cell design

    commit 47e57bbe87726e2756a7e75d0c917b8981d4320a
    Author: RaniaZeid <67391043+RaniaZeid@users.noreply.github.com>
    Date:   Thu Jul 28 14:25:28 2022 -0700

        create reviwer Table View Nib file

    commit e799b756eff230a3cb8cb9fc96b3be26796ccfbc
    Author: claramounir <claramounir555@gmail.com>
    Date:   Thu Jul 28 19:47:24 2022 +0200

        restore reviews view

    commit 663a7dee17fbb3b217c2532ad4b035a68d49afa8
    Author: claramounir <claramounir555@gmail.com>
    Date:   Thu Jul 28 19:34:11 2022 +0200

        solve review notes

    commit 68c7eb8358218c23593362abe14d9f4677d85a84
    Merge: 597352ee 575cf24d
    Author: Ahmed M. Hassan <ahmdmhasn@gmail.com>
    Date:   Mon Jul 25 02:20:16 2022 +0200

        Merge remote-tracking branch 'origin/develop' into feat/product-details-reviews-view

        # Conflicts:
        #	Dokan/Dokan.xcodeproj/project.pbxproj

    commit 597352ee002ef1e2f12a445df801d6994bbfcc8b
    Author: claramounir <107275541+claramounir@users.noreply.github.com>
    Date:   Thu Jul 21 18:01:49 2022 +0200

        Update ReviewsView.swift

    commit b63ce257da8ce17c01d355399177ce97321e7704
    Author: claramounir <claramounir555@gmail.com>
    Date:   Wed Jul 20 21:45:40 2022 +0200

        add review view design  :v:

    commit 8312a23a121efe491e164a7abf550eef85747787
    Merge: ea116fc5 de202755
    Author: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>
    Date:   Sun Jul 17 11:07:55 2022 +0200

        Merge branch 'develop' into feat/updateswiftversionto5.5

    commit ea116fc52ff9e142673c1da8d751cc4cbc061488
    Merge: 6f035770 a581bb94
    Author: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>
    Date:   Sun Jul 17 11:02:47 2022 +0200

        Merge branch 'develop' into feat/updateswiftversionto5.5

    commit 6f035770ba83a29d83ec2b3297677f1a800e6bda
    Merge: d751c91e 5cfba698
    Author: RaniaZeid <67391043+RaniaZeid@users.noreply.github.com>
    Date:   Sat Jul 16 20:33:47 2022 +0200

        Merge pull request #60 from motoon-eg/feat/updateverificationscene

        add action to resend and mainbutton

    commit 5cfba698e79661ef02762f8829eae1ff1d3a585c
    Author: RaniaZeid <67391043+RaniaZeid@users.noreply.github.com>
    Date:   Sat Jul 16 11:31:19 2022 -0700

        add action to resend and mainbutton

    commit d751c91eaaca210a61d78a43483e51683f5809fb
    Author: Ahmad Yasser <42544598+ahmdyasser@users.noreply.github.com>
    Date:   Sat Jul 16 19:41:00 2022 +0200

        make OTCTextfield class open (#59)

    commit b3339a93105e9e98fae67c96f2486bbb4d7e783f
    Author: RaniaZeid <67391043+RaniaZeid@users.noreply.github.com>
    Date:   Sat Jul 16 06:40:49 2022 -0700

        update swift version at domain and core

commit fa079d761cabe20bd96f49a48704865f8ba4b963
Author: MarwaEbrahem <marwa15shanab@gmail.com>
Date:   Sun Sep 4 16:09:07 2022 -0700

    fix: refactor MockNetwork

commit 40d4b7386ae0a5b5a27d4ac747ed7af4a0767049
Author: MarwaEbrahem <marwa15shanab@gmail.com>
Date:   Sun Sep 4 13:37:01 2022 -0700

    fix: update register ReviewerTableViewCell

commit 2e171d123c70a2a3a8badcf8de8e5d069f9ab02a
Author: MarwaEbrahem <marwa15shanab@gmail.com>
Date:   Sun Sep 4 12:47:39 2022 -0700

    fix: refactor ReviewProductNavigationBarBehavior

commit 20af4f09e88d1b53fdb046a2e6331b6a7113db2f
Author: MarwaEbrahem <marwa15shanab@gmail.com>
Date:   Sun Sep 4 12:47:01 2022 -0700

    fix: refactor RatingDetailsViewModel

commit 2bcbcfd23775782ee7ccc48661112456ee45857d
Author: MarwaEbrahem <marwa15shanab@gmail.com>
Date:   Sun Sep 4 12:30:37 2022 -0700

    fix: Update ReviewerTableViewCell

commit 399858c92a705a0b204ed04c7346609db5153d23
Author: MarwaEbrahem <marwa15shanab@gmail.com>
Date:   Sat Sep 3 10:36:27 2022 -0700

    fix: update code

commit d89c97ea9d30d1d10e4d062606b85a0e74f0b8bb
Author: MarwaEbrahem <31275340+MarwaEbrahem@users.noreply.github.com>
Date:   Fri Sep 2 01:00:39 2022 +0200

    fix: update state enum name

    Co-authored-by: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>

commit 564f14db5df170ca5960505c40209d24dc3a0137
Author: MarwaEbrahem <31275340+MarwaEbrahem@users.noreply.github.com>
Date:   Fri Sep 2 00:57:59 2022 +0200

    Update Dokan/Dokan/Scenes/ReviewProduct/ReviewProductNavigationBarBehavior.swift

    Co-authored-by: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>

commit b4c4601a25dd2a32ca9e1ac7423eb624943f1049
Author: MarwaEbrahem <31275340+MarwaEbrahem@users.noreply.github.com>
Date:   Fri Sep 2 00:57:11 2022 +0200

    fix: Update initialize of navBarRating variable

    Co-authored-by: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>

commit 29e0f850104c15a17be541a1628ee0779d370533
Author: MarwaEbrahem <31275340+MarwaEbrahem@users.noreply.github.com>
Date:   Fri Sep 2 00:54:58 2022 +0200

    fix: Update initialize of alertMessage variable

    Co-authored-by: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>

commit 083d01ffbf769b37fc99371613cb03789233ef65
Author: MarwaEbrahem <31275340+MarwaEbrahem@users.noreply.github.com>
Date:   Fri Sep 2 00:52:24 2022 +0200

    fix: Update initialize of cellViewModels variable

    Co-authored-by: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>

commit f2d4324a329bd9151d16927a944bbe90343a8b92
Author: MarwaEbrahem <31275340+MarwaEbrahem@users.noreply.github.com>
Date:   Fri Sep 2 00:49:36 2022 +0200

    fix: Update initialize of reviewList variable

    Co-authored-by: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>

commit 745ba58681a425dbaf19e0b10dbd525f4307af11
Author: MarwaEbrahem <31275340+MarwaEbrahem@users.noreply.github.com>
Date:   Fri Sep 2 00:42:11 2022 +0200

    fix: edit processFetchedReviews function

    Co-authored-by: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>

commit daadc306c618c9778425e79f8d0dea294905a720
Author: MarwaEbrahem <31275340+MarwaEbrahem@users.noreply.github.com>
Date:   Fri Sep 2 00:34:36 2022 +0200

    Update Reviews init function style

    Co-authored-by: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>

commit 2d17bf1b342b1c7aac4cbe7e240dad2a38e83be4
Author: MarwaEbrahem <31275340+MarwaEbrahem@users.noreply.github.com>
Date:   Fri Sep 2 00:17:13 2022 +0200

    fix: change codable to decodable in model

    Co-authored-by: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>

commit 19626e51dceab40c2e724f39e4dcd08bf0e70173
Author: MarwaEbrahem <31275340+MarwaEbrahem@users.noreply.github.com>
Date:   Fri Sep 2 00:15:32 2022 +0200

    fix: edit toDomain function

    Co-authored-by: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>

commit f9644ceb638a5b724412bb533d588ec454350c5a
Author: RaniaZeid <67391043+RaniaZeid@users.noreply.github.com>
Date:   Thu Sep 1 13:31:29 2022 -0700

    feat: Get All Categories List to Category Remote  (#114)

    * create all category remote file

    * rename all category protocol and class file

commit 840d226f9d574a6f950f6ef745528988a0ce56d8
Author: Ahmed Nagy <ahmednaagy97@gmail.com>
Date:   Sun Aug 21 23:23:23 2022 +0200

    fix: buttonsView compatibility with iOS 13 and above 📈 (#106)

    * add buttons view

    * fix ButtonsView not showing and change the favorite button color

    * delete the test code in SceneDelegate

    * change favorite button color

    * make product details screen the root viewController

    code added to SceneDelegate to launch the Product Details screen on launch to test out current tasks

    * fix: make buttonsView compatible starting from iOS 13 and above

    * adding access control to ButtonsView

    * resolving minor issues regarding last PR comments

    * fixing naming issues

    * fix: sellerImageProfile outlets connections

    * fix InfoSellerView outlets connections

commit e9d38ab9abb078b4bda2eef38aea9f8036524862
Merge: ed39eef3 21d6b271
Author: claramounir <107275541+claramounir@users.noreply.github.com>
Date:   Sat Aug 20 13:11:34 2022 +0200

    Merge pull request #65 from motoon-eg/feat/product-details-reviews-view

    feat: Add reviews section to product details page 🌠

commit 21d6b2711ba45a4805cc9a4d04cf6056a626a592
Merge: f6ff7837 ed39eef3
Author: claramounir <claramounir555@gmail.com>
Date:   Sat Aug 20 12:53:53 2022 +0200

    Merge branch 'develop' into feat/product-details-reviews-view

    # Conflicts:
    #	Dokan/Dokan.xcodeproj/project.pbxproj
    #	Dokan/Dokan/Scenes/ProductDetails/Views/ReviewerTableViewCell/ReviewerTableViewCell.swift

commit f6ff783792814f51189c47d9ab71eca89541e70e
Author: claramounir <claramounir555@gmail.com>
Date:   Fri Aug 19 18:11:06 2022 +0200

    add reviews view

commit 3783ebd324935df55a5a9d05aa1f9fed2d38b1cb
Author: claramounir <claramounir555@gmail.com>
Date:   Fri Aug 19 18:07:21 2022 +0200

    add UIView+Extension

commit 8b9f9218e68f30efc5d945a49d12a570d3f3551b
Author: claramounir <claramounir555@gmail.com>
Date:   Sun Aug 7 01:30:24 2022 +0200

    delete duplicate files

commit daeb336a950df7c9ca81c1108b7522a5902a13ad
Merge: c8dbddcf 6a5cbfbc
Author: claramounir <claramounir555@gmail.com>
Date:   Sun Aug 7 00:39:31 2022 +0200

    Merge branch 'feat/product-details-reviews-view' of https://github.com/motoon-eg/dokan-store into feat/product-details-reviews-view

commit c8dbddcf4008cc2696b4b784944a87db826a4ae1
Author: claramounir <claramounir555@gmail.com>
Date:   Sun Aug 7 00:35:53 2022 +0200

    reslove duplicate build error

commit 6a5cbfbc707cee8e3c6bac92dd26bd844aca4677
Merge: d1817094 288c6c58
Author: claramounir <107275541+claramounir@users.noreply.github.com>
Date:   Sun Aug 7 00:16:08 2022 +0200

    Merge branch 'develop' into feat/product-details-reviews-view

commit d18170949d4ef49d0167f851c2586b39341c23ed
Author: claramounir <claramounir555@gmail.com>
Date:   Sun Aug 7 00:04:48 2022 +0200

    solve conflict

commit fd77eb5a9566b220d7296deb1afababbdea27535
Author: claramounir <107275541+claramounir@users.noreply.github.com>
Date:   Sat Aug 6 07:58:23 2022 +0200

    Update ReviewsView.swift

commit 5e6d6554a6e05d3771a8907681473c136e349f7e
Author: claramounir <107275541+claramounir@users.noreply.github.com>
Date:   Sat Aug 6 07:54:11 2022 +0200

    Update Dokan/Dokan/Scenes/ProductDetails/ReviewsView.swift

    Co-authored-by: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>

commit ab944f9313c4c366ea6696039c128151e68f55a6
Author: claramounir <107275541+claramounir@users.noreply.github.com>
Date:   Sat Aug 6 07:53:50 2022 +0200

    Update Dokan/Dokan/Scenes/ProductDetails/Cell/ReviewerTableViewCell.swift

    Co-authored-by: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>

commit 8a6581d8b2bdbc4759183203ff657d5a4ccb71bb
Author: claramounir <107275541+claramounir@users.noreply.github.com>
Date:   Sat Aug 6 07:52:07 2022 +0200

    Update ReviewerTableViewCell.swift

commit e5043993868446deb50cffe483096ed0b8d8063d
Author: claramounir <107275541+claramounir@users.noreply.github.com>
Date:   Sat Aug 6 07:45:26 2022 +0200

    Update ReviewerTableViewCell.swift

commit 068aa17882e14ccf5e9f53c917e8d25cd3918c92
Author: claramounir <107275541+claramounir@users.noreply.github.com>
Date:   Sat Aug 6 07:38:54 2022 +0200

    Update Dokan/Dokan/Scenes/ProductDetails/Cell/ReviewerTableViewCell.swift

    Co-authored-by: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>

commit de18a8ffb105e852a4b309ff3ed650e30d752b8f
Author: claramounir <107275541+claramounir@users.noreply.github.com>
Date:   Sat Aug 6 07:38:34 2022 +0200

    Update ReviewerTableViewCell.swift

commit e7266756d96082c2d741b4aca2bbc16666914f0c
Author: claramounir <107275541+claramounir@users.noreply.github.com>
Date:   Sat Aug 6 07:37:30 2022 +0200

    Update ReviewerTableViewCell.swift

commit f50ab4dc7ea6643c9b23571e36df582751fc592e
Merge: 73b6bcf3 6ba5dca3
Author: claramounir <107275541+claramounir@users.noreply.github.com>
Date:   Sat Aug 6 07:35:20 2022 +0200

    Merge branch 'develop' into feat/product-details-reviews-view

commit 73b6bcf3f672e177bd9725107fef4bcb009e8170
Author: Ahmed M. Hassan <ahmdmhasn@gmail.com>
Date:   Sat Aug 6 01:35:59 2022 +0200

    Fix indentation

commit 3a4a656389d37b36f699a62e70cadafeec9ebf16
Author: Ahmed M. Hassan <ahmdmhasn@gmail.com>
Date:   Sat Aug 6 01:31:16 2022 +0200

    Fix conflicts

commit e839b0a4f635a2a85c948fd405a9953a65075ca9
Author: claramounir <claramounir555@gmail.com>
Date:   Sun Jul 31 22:41:26 2022 +0200

    ui changes

commit 786850b3b40ddba0fa46ac509e2aae3ef9dd2a1b
Author: RaniaZeid <67391043+RaniaZeid@users.noreply.github.com>
Date:   Thu Jul 28 14:25:28 2022 -0700

    create reviwer Table View Nib file

commit 437cbea8dc6d42989210af2272348f25bc795adc
Author: claramounir <claramounir555@gmail.com>
Date:   Thu Jul 28 19:47:24 2022 +0200

    restore reviews view

commit 73289b504bacaffa01afaf4838a41e71794b17f8
Author: claramounir <claramounir555@gmail.com>
Date:   Thu Jul 28 19:34:11 2022 +0200

    solve review notes

commit 514382c2d4bc4522dc59828d386b8b3d896f73ad
Author: claramounir <claramounir555@gmail.com>
Date:   Wed Jul 20 21:45:40 2022 +0200

    add review view design  :v:

commit ab0ee749f1ff5ecc74e8a834c4a94cc78a2d19f5
Merge: 34ddf7e7 c08c5b29
Author: Ahmed M. Hassan <ahmdmhasn@gmail.com>
Date:   Sat Aug 6 01:11:29 2022 +0200

    Merge remote-tracking branch 'origin/develop' into feat/product-details-reviews-view

    # Conflicts:
    #	Dokan/Dokan/Assets.xcassets/star.imageset/star.png

commit 34ddf7e74a4b4ec4e449e6a56be0eaadcfd470b7
Merge: cfe40eb5 a12ae42a
Author: claramounir <107275541+claramounir@users.noreply.github.com>
Date:   Thu Aug 4 12:42:29 2022 +0200

    Merge branch 'develop' into feat/product-details-reviews-view

commit cfe40eb5961904b28f74893738703cc2325a9b2e
Merge: e0847892 d258362d
Author: claramounir <107275541+claramounir@users.noreply.github.com>
Date:   Wed Aug 3 22:07:53 2022 +0200

    Merge branch 'develop' into feat/product-details-reviews-view

commit e0847892463df34b5ca886cf9ca15a2fc2b73976
Author: claramounir <claramounir555@gmail.com>
Date:   Tue Aug 2 16:12:33 2022 +0200

    configure review cell

commit 3023d59314659dbd92b6e1e5f1eb73acad0f0123
Author: claramounir <claramounir555@gmail.com>
Date:   Sun Jul 31 22:41:26 2022 +0200

    ui changes

commit 9d1556dfe6b7b53213bbf21fa59004e8e92e2b4b
Author: claramounir <claramounir555@gmail.com>
Date:   Fri Jul 29 15:53:59 2022 +0200

    solve conflict

commit c2f35cf71523af33cf1c0226e5505b750b51fdb8
Merge: 3845ac22 f661127e
Author: claramounir <claramounir555@gmail.com>
Date:   Fri Jul 29 01:09:18 2022 +0200

    no message

commit 3845ac2233487d5387359320ce4f461ee28d0d16
Author: claramounir <claramounir555@gmail.com>
Date:   Fri Jul 29 01:03:28 2022 +0200

    reviews view tests

commit f661127e4309b5673bd3f69d28dce1a6a9b6c399
Author: RaniaZeid <67391043+RaniaZeid@users.noreply.github.com>
Date:   Thu Jul 28 14:50:21 2022 -0700

    add outlets and cell model

commit 307a54e280117966641dd2f309fc3ecbe8bbcaf9
Author: RaniaZeid <67391043+RaniaZeid@users.noreply.github.com>
Date:   Thu Jul 28 14:44:36 2022 -0700

    create reviewer cell design

commit 47e57bbe87726e2756a7e75d0c917b8981d4320a
Author: RaniaZeid <67391043+RaniaZeid@users.noreply.github.com>
Date:   Thu Jul 28 14:25:28 2022 -0700

    create reviwer Table View Nib file

commit e799b756eff230a3cb8cb9fc96b3be26796ccfbc
Author: claramounir <claramounir555@gmail.com>
Date:   Thu Jul 28 19:47:24 2022 +0200

    restore reviews view

commit 663a7dee17fbb3b217c2532ad4b035a68d49afa8
Author: claramounir <claramounir555@gmail.com>
Date:   Thu Jul 28 19:34:11 2022 +0200

    solve review notes

commit 68c7eb8358218c23593362abe14d9f4677d85a84
Merge: 597352ee 575cf24d
Author: Ahmed M. Hassan <ahmdmhasn@gmail.com>
Date:   Mon Jul 25 02:20:16 2022 +0200

    Merge remote-tracking branch 'origin/develop' into feat/product-details-reviews-view

    # Conflicts:
    #	Dokan/Dokan.xcodeproj/project.pbxproj

commit 597352ee002ef1e2f12a445df801d6994bbfcc8b
Author: claramounir <107275541+claramounir@users.noreply.github.com>
Date:   Thu Jul 21 18:01:49 2022 +0200

    Update ReviewsView.swift

commit b63ce257da8ce17c01d355399177ce97321e7704
Author: claramounir <claramounir555@gmail.com>
Date:   Wed Jul 20 21:45:40 2022 +0200

    add review view design  :v:

commit 8312a23a121efe491e164a7abf550eef85747787
Merge: ea116fc5 de202755
Author: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>
Date:   Sun Jul 17 11:07:55 2022 +0200

    Merge branch 'develop' into feat/updateswiftversionto5.5

commit ea116fc52ff9e142673c1da8d751cc4cbc061488
Merge: 6f035770 a581bb94
Author: Ahmed M. Hassan <45182214+ahmdmhasn@users.noreply.github.com>
Date:   Sun Jul 17 11:02:47 2022 +0200

    Merge branch 'develop' into feat/updateswiftversionto5.5

commit 6f035770ba83a29d83ec2b3297677f1a800e6bda
Merge: d751c91e 5cfba698
Author: RaniaZeid <67391043+RaniaZeid@users.noreply.github.com>
Date:   Sat Jul 16 20:33:47 2022 +0200

    Merge pull request #60 from motoon-eg/feat/updateverificationscene

    add action to resend and mainbutton

commit 5cfba698e79661ef02762f8829eae1ff1d3a585c
Author: RaniaZeid <67391043+RaniaZeid@users.noreply.github.com>
Date:   Sat Jul 16 11:31:19 2022 -0700

    add action to resend and mainbutton

commit d751c91eaaca210a61d78a43483e51683f5809fb
Author: Ahmad Yasser <42544598+ahmdyasser@users.noreply.github.com>
Date:   Sat Jul 16 19:41:00 2022 +0200

    make OTCTextfield class open (#59)

commit b3339a93105e9e98fae67c96f2486bbb4d7e783f
Author: RaniaZeid <67391043+RaniaZeid@users.noreply.github.com>
Date:   Sat Jul 16 06:40:49 2022 -0700

    update swift version at domain and core

* feat: Add mockReviewProductViewModel file
